vyper/semantics/validation/annotation.py

```diff
@@ -198,20 +223,26 @@ def visit_Name(self, node, type_):
         node._metadata["type"] = get_exact_type_from_node(node)
 
     def visit_Subscript(self, node, type_):
-        if isinstance(node.value, vy_ast.List):
-            possible_base_types = get_possible_types_from_node(node.value)
+        # Set base_type to None if it is a typestring
+        if type_ and not isinstance(type_, BaseTypeDefinition):
+            base_type = None
```

Comment:
It is used for early return at L247: https://github.com/vyperlang/vyper/blob/fc18d87fa7bb6bcc9a9fffd1ab9b86ae968ad882/vyper/semantics/validation/annotation.py#L247

Reply:
ah, missed that in the diff. i think, instead of having a special `base_type` value to trigger the fastpath later in the function, it will be better to move the fastpath return directly into this branch.